crates/sui-node/src/lib.rs

```diff
@@ -1657,15 +1659,19 @@ impl SuiNode {
                 consensus_store_pruner,
                 consensus_adapter,
                 checkpoint_service_exit,
+                mut checkpoint_service_tasks,
                 checkpoint_metrics,
                 sui_tx_validator_metrics,
             }) = self.validator_components.lock().await.take()
             {
                 info!("Reconfiguring the validator.");
-                // Stop the old checkpoint service.
-                drop(checkpoint_service_exit);
+                // Stop the old checkpoint service and wait for them to finish.
+                let _ = checkpoint_service_exit.send(());
+                while let Some(_result) = checkpoint_service_tasks.join_next().await {}
```

Comment:
Do we want to add a timeout here in case something is wrong in the service to prevent it from stopping?

Reply:
Added a panic in tests if this times out.